subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/SwiftCompile.java

```diff
@@ -78,4 +80,18 @@ public void setModuleName(String moduleName) {
     public void setModuleName(Provider<String> moduleName) {
         this.moduleName.set(moduleName);
     }
+
+    @Override
+    public void compile(IncrementalTaskInputs inputs) {
+        SimpleStaleClassCleaner cleaner = new SimpleStaleClassCleaner(getOutputs());
```

Comment:
Isn't this just a fancy way of deleting all of the outputs every time?

There's this TODO:
https://github.com/gradle/gradle/blob/85c23b8dc5a1edfe0e957c041f24d71179bbe792/subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/swift/SwiftPlatformToolProvider.java#L73-L74

Is this related? Shouldn't we already be deleting stale outputs?

Reply:
That `OutputCleaningCompiler` won't actually do anything, because there's nothing upstream of it calculating which object files need to be removed, which is what it's expecting. It just shouldn't be there at all.

Reply:
Adam is right on this, we shortcut the incremental compiler creation by returning the normal, non-incremental, compiler. The issue is `IncrementalNativeCompiler` will do the wrong thing as it's opinionated toward `#include` processing. This will be addressed when we roll incremental support for Swift. Right now, we just want the stale object file to be cleaned up.